File: leetcode/completed/303.java
public class 303 {
    class NumArray {

        int[] prefix;
    
        public NumArray(int[] nums) {
            prefix = new int[nums.length+1];
            for (int i = 1; i < nums.length+1; i++) {
                prefix[i] = prefix[i-1] + nums[i-1];
            }
        }
        
        public int sumRange(int left, int right) {
            return prefix[right+1] - prefix[left];
        }
    }
    
    /**
     * Your NumArray object will be instantiated and called as such:
     * NumArray obj = new NumArray(nums);
     * int param_1 = obj.sumRange(left,right);
     */
}

File: leetcode/completed/303.cpp
#include <vector>

using namespace std;

class NumArray {
public:

    vector<int> prefix = {0};

    NumArray(vector<int>& nums) {
        for (int i = 1; i < nums.size()+1; i++) {
            prefix.push_back(prefix[i-1] + nums[i-1]);
        }
    }
    
    int sumRange(int left, int right) {
        return prefix[right+1] - prefix[left];
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */